server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java

```diff
@@ -1438,4 +1450,161 @@ public void testRestartPrimary_NoReplicas() throws Exception {
         ensureYellow(INDEX_NAME);
         assertDocCounts(1, primary);
     }
+
+    /**
+     * Tests whether segment replication supports realtime get requests and reads and parses source from the translog to serve strong reads.
+     */
+    public void testRealtimeGetRequestsSuccessful() {
+        final String primary = internalCluster().startDataOnlyNode();
+        final String replica = internalCluster().startDataOnlyNode();
+
+        assertAcked(
+            prepareCreate(INDEX_NAME).setSettings(Settings.builder().put("index.refresh_interval", -1).put(indexSettings()))
+                .addAlias(new Alias("alias"))
+        );
+        ensureGreen(INDEX_NAME);
+
+        GetResponse response = client().prepareGet(indexOrAlias(), "1").get();
+        assertFalse(response.isExists());
+
+        // index doc 1
+        client().prepareIndex(indexOrAlias()).setId("1").setSource("foo", "bar").get();
+
+        // non realtime get 1
+        response = client().prepareGet(indexOrAlias(), "1").setRealtime(false).get();
+        assertFalse(response.isExists());
+
+        // non realtime get 1 (on replica shard only)
+        response = client(replica).prepareGet(indexOrAlias(), "1").setPreference("_only_local").setRealtime(false).get();
+        assertFalse(response.isExists());
+
+        // realtime get 1
+        response = client().prepareGet(indexOrAlias(), "1").get();
+        assertTrue(response.isExists());
+        assertThat(response.getIndex(), equalTo(INDEX_NAME));
+        assertThat(response.getSourceAsMap().get("foo").toString(), equalTo("bar"));
+    }
+
+    public void testRealtimeGetRequestsUnsuccessful() {
+        final String primary = internalCluster().startDataOnlyNode();
+        final String replica = internalCluster().startDataOnlyNode();
+
+        assertAcked(
```

Comment:
In this test and in the other, I think we need to create the index after creating the primary node to guarantee which node holds which shard.  Otherwise either can contain the primary.